Feature/storybook

b12f opened the pull request:
This installs storybook and adds a docker container for it.

The deleted component file was apparently a leftover that should've been deleted earlier. It broke the storybook build because the template import was not available.

Still includes the default stories that storybook installs.

Relates to: https://community.openproject.org/work_packages/41847
Relates to: https://community.openproject.org/projects/openproject/work_packages/33242/

The `dev` branch documentation is visible at https://opf.github.io/design-system/

Commit:
Initial storybook commit

Commit:
Fix documentation.json links

oliverguenther commented:
Only had the time for a short glimpse yet, but a few things popped up:

1. We don't want to commit the documentation.json, as it is being rebuild on every serve and will be impossible to merge automatically.
2. I'd expect the first merged version to include a documentation of at least part of the spot components. This certainly doesn't have to be complete, but we shouldn't merge the default examples
3. The past times I've tried storybook, using an angular component from within mdx would always fail for mysterious reasons. It's been a while since I tried, but I'd like to see that working before we continue

oliverguenther commented:
Only had the time for a short glimpse yet, but a few things popped up:

1. We don't want to commit the documentation.json, as it is being rebuild on every serve and will be impossible to merge automatically.
2. I'd expect the first merged version to include a documentation of at least part of the spot components. This certainly doesn't have to be complete, but we shouldn't merge the default examples
3. The past times I've tried storybook, using an angular component from within mdx would always fail for mysterious reasons. It's been a while since I tried, but I'd like to see that working before we continue

Commit:
Don't track documentation.json in git

Commit:
Enable sass in storybook

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Initial version of a story that uses angular components

Commit:
Remove example stories, clean up button story

Commit:
More example stories

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Fix sb build

Commit:
Always use dev

Commit:
Try without auth header

Commit:
Update workflow name

Commit:
More logs

Commit:
Check if token set

Commit:
Use release/storybook branch for testing

Commit:
Send ref input to workflow

Commit:
Escape input to curl call

Commit:
Adding logging

Commit:
Different type of escaping

Commit:
Fix JOSN

Commit:
Use dev branch for opf/design-system storybook publishing

Commit:
Add plugin to message path to parent window

Commit:
Remove extraneous story

Commit:
Add a ton of docs

Commit:
Update stories

Commit:
Fix syntax error caused by multiple newlines inside of a JSX component

Commit:
Add text-field story

Commit:
Add basic html stories that don't work yet

Commit:
Try to get plain HTML examples working

Commit:
HTML Examples work, but slowly revert to components anyway

Commit:
Fix HTML examples

Commit:
Remove extraneous files

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Put storybook eslint rules back in

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Improve docs

Commit:
Merge branch 'feature/storybook' of github.com:opf/openproject into feature/storybook

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Show docs tab by default

Commit:
Add pullpreview for storybook

Commit:
Use the same pullpreview tag for both storybook and normal deployments

Commit:
Merge branch 'dev' into feature/storybook

Commit:
Change name of second pullpreview workflow

Commit:
Pin node version to 16.17.0

psatyal committed:
Initial update to docs

Added/updated:

Foundation pages:

- Colours (major update)
- Shadows (minor)
- Typography (new)

Blocks

- Checkbox (minor)
- Action bar (major)
- Buttons (new)
- Link (major)
- Modal Dialogue (new)
- Selector Field (new)

Commit:
Make sure all code is available during storybook pp build

Commit:
Change storybook pullpreview file name

Commit:
Add production target to docker-compose sb pp

Commit:
Fix acme check for sb pp

Commit:
Only run cd-storybook on dev branch

Commit:
Run without https on 8080

psatyal committed:
Added intro and new page

- Introduction renamed to "Design System", page rewritten completely
- Added page "Devices and Accessibility"

Commit:
Remove domain from caddy

Commit:
Merge branch 'feature/storybook' of github.com:opf/openproject into feature/storybook

Commit:
Add port to listen command

Commit:
Remove double pullpreview workflows

psatyal committed:
Added Divider component

Commit:
Change sorting of stories

Commit:
Merge branch 'feature/storybook' of github.com:opf/openproject into feature/storybook

Commit:
Merge branch 'dev' into feature/storybook

psatyal committed:
Update section titles and order for styles and blocks

Commit:
add extra action bar story

psatyal committed:
Updated organising + new page

- Updated organisation into Styles, Components and Patterns.
- Added page "Using Storybook" (mostly a skeleton for now)

psatyal committed:
Added note about colours not being implemented yet

psatyal committed:
Minor

Commit:
Merge branch 'dev' into feature/storybook
